### ISSUE TITLE
browser authorization url shows "401 Error: deleted_client"

### ISSUE THREAD

**Issue by yokomotod** [dask/gcsfs#171]
Hi, I faced issue that browser authorization url shows :

> 401. That’s an error.
> Error: deleted_client
> The OAuth client was deleted.

[screenshot]
![image](https://user-images.githubusercontent.com/825732/64016017-d73e6400-cb60-11e9-96c1-f35f84223aa5.png)

[reproduce]

```
pipenv --python 3.7.4
pipenv install gcsfs
```

```
import gcsfs

fs = gcsfs.GCSFileSystem(token="browser")
```

```
Please visit this URL to authorize this application: https://accounts.google.com/o/oauth2/auth?response_type=code&client_id=586241054156-8986sjc0h0683jmpb150i0m8cucrttds.apps.googleusercontent.com&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdevstorage.full_control&state=TIrQYtqv2eKxHpW8CL0xKgm3JDl626&prompt=consent&access_type=offline&code_challenge=eo357tajzJNw6Nev0YwigXfZU-_32IqJyQ9nNCWcyfM&code_challenge_method=S256
```

**Comment by yokomotod** [comment]
If I change the committed [google oauth2 client id](https://github.com/dask/gcsfs/blob/master/gcsfs/core.py#L58)  to another one (which I created by myself) then It works.

Is the committed client id really deleted ... ?? 😳

**Comment by martindurant** [comment]
I believe it must have expired. I'll try to get around to making another.

**Comment by martindurant** [comment]
I tried to create a new client also, but got the same error. How did you create the client that worked?

**Comment by leehart** [comment]
Our workaround was to override the client:
```
gcsfs.core.client_config['installed']['client_id'] = 'foobar.apps.googleusercontent.com'
gcsfs.core.client_config['installed']['client_secret'] = 'M4rt1nLik3sF00d'
```

AFAIK, our OAuth 2.0 client ID and "secret" was created through the GCP web GUI, https://console.cloud.google.com/apis/credentials

**Comment by yokomotod** [comment]
@martindurant 

I've created the client as

1. Go to Web Console: https://console.developers.google.com/apis/credentials
1. "Create credentials" > "OAuth client ID"
1. "Application type: Other"
1. Update https://github.com/dask/gcsfs/blob/master/gcsfs/core.py#L58

**Comment by chrisroat** [comment]
Please consider a new release, thank you!

**Comment by TomAugspurger** [comment]
@martindurant do you think master is releasable? If so, I can cut a release
today sometime.

**Comment by martindurant** [comment]
Yes, together with fsspec as always, but https://github.com/intake/filesystem_spec/pull/146 and https://github.com/dask/s3fs/pull/240 are of interest in any release cycle.
There are also the ongoing intermittent errors, but that can be another release when they are fixed.

**Comment by TomAugspurger** [comment]
So gcsfs master needs fsspec master right now? gcsfs, fsspec, and s3fs
should all be released simultaneously?

**Comment by martindurant** [comment]
> gcsfs, fsspec, and s3fs should all be released simultaneously

I am not totally certain, but seems like good policy

**Comment by TomAugspurger** [comment]
OK, I'll try to do a batch of releases on Monday then (traveling tomorrow).

**Comment by martindurant** [comment]
In progress now, conda-forge will be with us shortly.
